### ISSUE TITLE
Many complex types do not implement the `std::fmt::Debug` trait

### ISSUE THREAD

**Issue by burtonageo** [Adanos020/egui_dock#82]
**Is your feature request related to a problem? Please describe.**
When embedding `egui_dock` types in other types in my code, I cannot use the `#[derive(Debug)]` attribute on them as they themselves do not implement the `Debug` trait.

**Describe the solution you'd like**
All `egui_dock` types should implement the `Debug` trait. For complex types, an opaque debug impl should be acceptable, e.g:

```rust
use std::fmt;

impl<Tab> fmt::Debug for Tree<Tab> {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        f.debug_struct("Tree").finish_non_exhaustive()
    }
}
```

**Describe alternatives you've considered**
The obvious alternative is to add a custom `Debug` impl and skip over the `egui_dock` types embedded in my types, but this would add extra code. Another alternative is to make my types not implement `Debug`, but this is inconvenient  when debugging, and there are areas in my code where generics enforce that a type implements `Debug`.

**Comment by burtonageo** [comment]
If this is something you think would be good to have in `egui_dock`, I'd be happy to implement and submit a pull request for this

**Comment by Adanos020** [comment]
Hey, this sounds good to me. If you want to implement this, feel free to go ahead!

**Comment by Adanos020** [comment]
Closed by #84

**Comment by blueforesticarus** [comment]
the Debug impl for Tree really ought to include the Tree. 
at the moment it just gives `Tree { .. }`
